### ISSUE TITLE
SSL routines:SSL_CTX_new:library has no ciphers:ssl/ssl_lib.c:2400

### ISSUE THREAD

**Issue by tengliu0929** [openssl/openssl#13296]
Hello, I've written a jni that call c dynamic lib which base on openssl 1.1.1b version. I wrote a c program to call the dynamic lib, it was success,but when I used java, this wrong message which encontered.
Otherwise, I 've initilized the openssl by using SSL_library_init and OpenSSL_add_ssl_algorithms.
Is there any way that can solve this problem? thank you

**Comment by mattcaswell** [comment]
Hmm. That's quite strange.

> Otherwise, I 've initilized the openssl by using SSL_library_init and OpenSSL_add_ssl_algorithms.

This shouldn't be necessary in 1.1.1. All the ciphers should be added automatically.

Is there any possibility you can compile OpenSSL with debug symbols and step through it in a debugger to see where it is going wrong?

**Comment by tengliu0929** [comment]
Does the version which had been modified to support the Chinese TLS version matters?

**Comment by mattcaswell** [comment]
> Does the version which had been modified to support the Chinese TLS version matters?

Do I understand correctly that you are using a patched version of OpenSSL for Chinese TLS support?? If so then this almost certainly matters. You should raise your issue against whatever project provided the patches in the first instance. Or alternatively try your code against an unpatched version of OpenSSl and see if you get the same problem.

**Comment by tengliu0929** [comment]
Yeah, surely a patch,but SSL_Ctx_new function was not modified here, and can not allocate memory for SSL_CTX struct. So weird phenomenon. Besides, thank you for answering.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of the 3.4 dev cycle, barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
